cmd/jaeger-v2/internal/components.go

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2023 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package internal
+
+import (
+	"github.com/open-telemetry/opentelemetry-collector-contrib/connector/spanmetricsconnector"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/kafkaexporter"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/jaegerreceiver"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/kafkareceiver"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/zipkinreceiver"
+	"go.opentelemetry.io/collector/connector"
+	"go.opentelemetry.io/collector/connector/forwardconnector"
+	"go.opentelemetry.io/collector/exporter"
+	"go.opentelemetry.io/collector/exporter/loggingexporter"
+	"go.opentelemetry.io/collector/exporter/otlpexporter"
+	"go.opentelemetry.io/collector/exporter/otlphttpexporter"
+	"go.opentelemetry.io/collector/extension"
+	"go.opentelemetry.io/collector/extension/ballastextension"
+	"go.opentelemetry.io/collector/extension/zpagesextension"
+	"go.opentelemetry.io/collector/otelcol"
+	"go.opentelemetry.io/collector/processor"
+	"go.opentelemetry.io/collector/processor/batchprocessor"
+	"go.opentelemetry.io/collector/processor/memorylimiterprocessor"
+	"go.opentelemetry.io/collector/receiver"
+	"go.opentelemetry.io/collector/receiver/otlpreceiver"
+
+	"github.com/jaegertracing/jaeger/cmd/jaeger-v2/internal/exporters/storageexporter"
+	"github.com/jaegertracing/jaeger/cmd/jaeger-v2/internal/extension/jaegerquery"
+	"github.com/jaegertracing/jaeger/cmd/jaeger-v2/internal/extension/jaegerstorage"
+)
+
+func components() (otelcol.Factories, error) {
+	var err error
+	factories := otelcol.Factories{}
+
+	factories.Extensions, err = extension.MakeFactoryMap(
+		// standard
+		ballastextension.NewFactory(),
+		zpagesextension.NewFactory(),
+		// add-ons
+		jaegerquery.NewFactory(),
+		jaegerstorage.NewFactory(),
+		// TODO add adaptive sampling
+	)
+	if err != nil {
+		return otelcol.Factories{}, err
+	}
+
+	factories.Receivers, err = receiver.MakeFactoryMap(
+		// standard
+		otlpreceiver.NewFactory(),
+		// add-ons
+		jaegerreceiver.NewFactory(),
+		kafkareceiver.NewFactory(),
+		zipkinreceiver.NewFactory(),
+	)
+	if err != nil {
+		return otelcol.Factories{}, err
+	}
+
+	factories.Exporters, err = exporter.MakeFactoryMap(
+		// standard
+		loggingexporter.NewFactory(),
+		otlpexporter.NewFactory(),
+		otlphttpexporter.NewFactory(),
+		// add-ons
+		storageexporter.NewFactory(), // generic exporter to Jaeger v1 spanstore.SpanWriter
+		kafkaexporter.NewFactory(),
+		// elasticsearch.NewFactory(),
```

Comment:
Why exclude the elasticsearch exporter? Is it because storageexporter should already cover the ability to write to elasticsearch using jaeger's native elasticsearch span writer?

Reply:
oh, because there isn't one, it's just the code I copied from Pavol's repo where es exporter module was defined, but was not implemented.

Initially yes, the generic support for existing storage backends is done via `storageexporter`, which converts OTLP into Jaeger model and sends to SpanWriter one span at a time. As we build OTLP-native exporters people can migrate to those.